core/src/main/scala/cats/UnorderedFoldable.scala

```diff
@@ -0,0 +1,66 @@
+package cats
+
+import cats.kernel.CommutativeMonoid
+import simulacrum.typeclass
+
+import scala.collection.mutable
+
+/**
+  * `UnorderedFoldable` is like a `Foldable` for unordered containers.
+  */
+@typeclass trait UnorderedFoldable[F[_]] {
+
+  /**
+    * Left associative fold on 'F' using the function 'f'.
+    */
+  def foldLeft[A, B](fa: F[A], b: B)(f: (B, A) => B): B
+
+  def unorderedFoldMap[A, B: CommutativeMonoid](fa: F[A])(f: A => B): B =
+    foldLeft(fa, Monoid[B].empty)((b, a) => Monoid[B].combine(b, f(a)))
+
+  def unorderedFold[A: CommutativeMonoid](fa: F[A]): A =
+    unorderedFoldMap(fa)(identity)
+
+  def toSet[A](fa: F[A]): Set[A] =
+    foldLeft(fa, mutable.ListBuffer.empty[A]) { (buf, a) =>
+      buf += a
+    }.toSet
+
+
+  /**
+    * Returns true if there are no elements. Otherwise false.
+    */
+  def isEmpty[A](fa: F[A]): Boolean =
```

Comment:
This can be implemented with foldMap (map to true, do the commutative “or” monoid